src/RowGateway/AbstractRowGateway.php

```diff
@@ -92,21 +95,12 @@ public function populate(array $rowData, $rowExistsInDatabase = false)
         return $this;
     }
 
-    /**
-     * @param mixed $array
-     * @return AbstractRowGateway
-     */
-    public function exchangeArray($array)
-    {
-        return $this->populate($array, true);
-    }
-
```

Comment:
Why is this being removed?

Reply:
This change was brought in from another PR that was merged in from the recent changes in 0.3.0.

I will finish that this evening and release a 3.1.0 so that @simon-mundy can merge those changes in. That was my oversight and not his. I was doing some work in RowGateway to improve consistency between it and the standardized ArrayObject defaults that back ResultSet in respect to its ObjectPrototype usage.